app/controllers/claims/application_controller.rb

```diff
@@ -1,3 +1,7 @@
 class Claims::ApplicationController < ApplicationController
   after_action :verify_authorized
+
+  def user_not_authorized
```

Comment:
Do you need this given the `user_not_authorized` in `ApplicationController`?

Reply:
> Do you need this given the `user_not_authorized` in `ApplicationController`? I can see you've made a comment in your PR description, see an example were you wouldn't get a referrer. If this is a concern for claims, this will be a concern for both services and perhaps should just be rescued with a redirect back

Reply:
My concern is that your change prevents the `redirect_back` completely within your service. I shouldn't redirect to the `root_path` in all case. Only in that specific use case where as you mention the `request.referrer` creates an infinite loop.

Reply:
https://medium.com/@infernalmaster/rails-infinite-redirect-problem-4a40d998e194